src/main/java/org/traccar/api/security/SecurityRequestFilter.java

```diff
@@ -116,4 +117,38 @@ public void filter(ContainerRequestContext requestContext) {
 
     }
 
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String authHeaders = httpRequest.getQueryString();
```

Comment:
Why can't we use headers?

Reply:
it seems that browsers don´t support sending headers with websocket, you can only send "protocols" or attach a cookie before calling the websocket
```
var ws = new WebSocket("ws://example.com/service", "soap");
```